web/main/models.py

```diff
@@ -4077,7 +4081,32 @@ class EmailWhitelist(models.Model):
 
 class LiveSettings(models.Model):
     prevent_exports = models.BooleanField(blank=False, default=False, null=False)
+    export_average_rate = models.IntegerField(blank=False, default=0)
+    export_last_minute_updated = models.IntegerField(blank=False, default=0)
 
+    @classmethod
+    @transaction.atomic
+    def export_is_rate_limited(cls):
+        """
+        >>> ls, full_casebook, resource = [getfixture(f) for f in ['live_settings','full_casebook','resource']]
+        >>> prior_count = ls.export_average_rate
+        >>> _ = full_casebook.export(False)
+        >>> _ = resource.export(False)
+        >>> ls.refresh_from_db()
+        >>> assert ls.export_average_rate == prior_count + 2
+        """
+        live_settings = LiveSettings.load()
+        current_time = datetime.now()
+        minute = current_time.hour*60 + current_time.minute
+        elapsed_minutes = (minute-live_settings.export_last_minute_updated) % 1440
+        new_rate = max(live_settings.export_average_rate - (elapsed_minutes * settings.EXPORT_RATE_FALLOFF), 0)
```

Comment:
This is a clever technique! I had to step through the math embarrassingly slowly on a note card to see how it works 😄.

Reply:
I drew a little diagram and then verified that mod arithmetic in python works the way I wanted it to (there are some languages where it doesn't, strangely).